app/src/main/java/com/lumisdinos/gameoffifteen/presentation/HomeViewModel.kt

```diff
@@ -30,4 +38,51 @@ class HomeViewModel @Inject constructor(
     }
 
 
+    private fun generateCells() {
+        digits.clear()
+        while (digits.size < 16) {
+            val digit = (0..15).random()
+            if (digits.contains(digit)) {
+                var isAdded = false
+                var nextDigit = digit
+                while (!isAdded) {
+                    nextDigit += 1
+                    if (nextDigit > 15) {
+                        nextDigit = 0
+                    }
+                    if (!digits.contains(nextDigit)) {
+                        digits.add(nextDigit)
+                        isAdded = true
+                    }
+                }
+            } else {
+                digits.add(digit)
+            }
+        }
+    }
```

Comment:
Mmm i do not understand this part. Are you creating list of numbers?
May be something like this:
val cells = (1..15).shuffled().take(15).toSet()

Reply:
Or this way: 
val cells = mutableSetOf<Int>()

while (cells .size != 15) {
    cells .add(Random.nextInt(1, 15))
}

Reply:
thanks, I use this one: 
val cells = (0..dimention).shuffled().toList()